stan/math/prim/fun/generalized_inverse.hpp

```diff
@@ -0,0 +1,101 @@
+#ifndef STAN_MATH_PRIM_FUN_GENERALIZED_INVERSE_HPP
+#define STAN_MATH_PRIM_FUN_GENERALIZED_INVERSE_HPP
+
+#include <stan/math/prim/meta.hpp>
+#include <stan/math/prim/err.hpp>
+#include <stan/math/prim/fun/Eigen.hpp>
+#include <stan/math/prim/fun/chol2inv.hpp>
+#include <stan/math/prim/fun/cholesky_decompose.hpp>
+#include <stan/math/prim/fun/transpose.hpp>
+#include <stan/math/prim/fun/tcrossprod.hpp>
+#include <stan/math/prim/fun/crossprod.hpp>
+#include <stan/math/prim/fun/quad_form.hpp>
+#include <stan/math/prim/fun/inverse_spd.hpp>
+#include <stan/math/prim/fun/mdivide_left_spd.hpp>
+#include <stan/math/prim/fun/mdivide_right_spd.hpp>
+#include <stan/math/prim/fun/inverse.hpp>
+
+namespace stan {
+namespace math {
+
+/**
+ * Returns the Moore-Penrose generalized inverse of the specified matrix.
+ *
+ * @tparam T type of elements in the matrix
+ * @tparam n number of rows, can be Eigen::Dynamic
+ * @tparam m number of columns, can be Eigen::Dynamic
```

Comment:
These need removed and docs for `EigMat` need to be added

Reply:
I updated but not sure if it's right, let me know.